homeassistant/components/octoprint/__init__.py

```diff
@@ -163,14 +163,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         data = {**entry.data, CONF_VERIFY_SSL: True}
         hass.config_entries.async_update_entry(entry, data=data)
 
-    verify_ssl = entry.data[CONF_VERIFY_SSL]
-    websession = async_get_clientsession(hass, verify_ssl=verify_ssl)
```

Comment:
Is this going to force ssl verification and ignore the setting?

Reply:
I don't know how this works, if we now need to set this on the session provided by `pyoctoprintapi`, then I guess it does.

Reply:
@rfleming71 Is this something that needs to be addressed before this PR can be reviewed again? If so please let me know how I can do this and what is neccesary. Otherwise I would open this PR for review again as is.

Reply:
We still need to consider the `CONF_VERIFY_SSL` configuration item, otherwise we'll break the integration for users that has changed it and need it.

When ready for a new review, please mark the PR as ready for review, using that button.

Reply:
How do I do this? Should I copy the contents of these two methods to `pyoctoprintapi`?

```
_DEFAULT_SSL_CONTEXT = client_context()
_DEFAULT_NO_VERIFY_SSL_CONTEXT = create_no_verify_ssl_context()
```

@rfleming71 its your library, would you be okay with me adding a `verify_ssl` parameter as well?

Reply:
I think you can simply accept a boolean parameter for `verify_ssl` in the client and if the parameter is `False` pass `ssl=False` to the connector when creating it.

https://docs.aiohttp.org/en/stable/client_reference.html#tcpconnector

Reply:
I am not sure I understand what you are suggesting. Initially I modified the HASS session, but I was asked to revert this and create the session inside of the lib instead. Now I should do it inside of HASS again? o.O

Reply:
Please read the aiohttp TCPConnector docs regarding `verify_ssl` and `ssl` parameters.

Reply:
@markusressel there is a difference between adjusting the helpers for every integration for your use case versus creating a one off session in the integration

Reply:
> Please read the aiohttp TCPConnector docs regarding verify_ssl and ssl parameters.

@MartinHjelmare Thx for the link, I have re-read the documentation. So if no custom certificate stuff is used the `ssl` and `verify_ssl` parameters of the `TCPConnector` serve pretty much the same purpose and by simply adding `verify_ssl` to the `pyoctoprintapi` client and passing this along as `ssl` parameter to the TCPConnector of the default session provided by the lib, it should work. Is this correct?

@balloob Thats correct, and we already discussed this and other options:

> Oh I think I misunderstood, you really only meant to not expand the helper, but still put the custom session into the octoprint package within home assistant, right?

I am really sorry if my communication was not good enough to explain or understand what is being suggested here. The limited length of messages does not help though :smiling_face_with_tear: I am happy to implement whatever is accepted by you guys, but I want to make sure I understand you, so I don't have to keep investing time into implementations that I can then revert again :smile:

At this point there are still two different approaches suggested, if I am not misunderstanding:

1. implement a custom session within the home assistant integration and pass it along
  * This solution seems to be preferred by @rfleming71 
2. add another parameter to the Client in `pyoctoprintapi` called `verify_ssl` and pass it along as `ssl` parameter to the internally created TCPConnector.

Which one shall it be?

Reply:
Btw, is this type of "mentioning" within a PR also bad and discouraged?
If so, I sincerely apologize, its how I do this at work and I didn't think about it until now.

Reply:
Ryan creates the API wrapper so I think it's his call. From my experience, `verify_ssl` is the only thing that's influencing session creation. I would not expect more parameters.

Mentioning is fine – I am already subscribed to this issue and get all notifications. Just don't do it for people not involved in the issue.

Reply:
@rfleming71 any thoughts on how to continue with this?

Reply:
I still lean towards option 1 given we never expect more parameters but they have a habit of cropping up 😄

Reply:
@rfleming71 :+1:
If there is no veto for this within the next few days from any of the Home Assistant maintainers, I will go ahead and implement this within the next 2 weeks. Just as an FYI: when I opened this PR, I had plenty of time to work on it, however, now that this discussion has taken a couple of weeks, thats not the case anymore, so please bear with me :heart: